ocaml/xapi-storage-script/main.ml

```diff
@@ -1553,9 +1553,13 @@ let bind ~volume_script_dir =
     @@
     let* sr = Attached_SRs.find sr in
     let vdi = Storage_interface.Vdi.string_of vdi in
-    return_volume_rpc (fun () ->
-        Volume_client.data_destroy volume_rpc dbg sr vdi
-    )
+    let* response =
+      return_volume_rpc (fun () ->
+          Volume_client.data_destroy volume_rpc dbg sr vdi
+      )
+    in
+    let* () = set ~dbg ~sr ~vdi ~key:_vdi_type_key ~value:"cbt_metadata" in
+    Deferred.Result.return response
```

Comment:
this `response` var is always going to be a unit, isn't it? We could just return unit here

Reply:
It could be changed to 
```ocaml
let* () = return_volume_rpc (...) in
set ~dbg ... ~value"cbt_metadata"
```

But the way it is right now is fine as well